### ISSUE TITLE
The model test failed！！！

### ISSUE THREAD

**Issue by gyxw521** [city-super/Scaffold-GS#53]
Great Work.
I am currently encountering a problem, the result of each test is NAN, the division of data train and test is unsuccessful, I can't find the reason, and there is no error, I hope you can answer it!!

**Run the results：**
Setting up [LPIPS] perceptual loss: trunk [vgg], v[0.1], spatial [off]
D:\conda\envs\Scaffold-GS\lib\site-packages\torchvision\models\_utils.py:208: UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.
  warnings.warn(
D:\conda\envs\Scaffold-GS\lib\site-packages\torchvision\models\_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=VGG16_Weights.IMAGENET1K_V1`. You can also use `weights=VGG16_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
Loading model from: D:\conda\envs\Scaffold-GS\lib\site-packages\lpips\weights\v0.1\vgg.pth
not found tf board
$CUDA_VISIBLE_DEVICES
Output folder: C://Users//lenovo//Desktop//XAG//daima//youhua//Scaffold-GS//output [28/05 20:25:56]
2024-05-28 20:25:56,921 - INFO: args: Namespace(add_color_dist=False, add_cov_dist=False, add_opacity_dist=False, appearance_dim=32, appearance_lr_delay_mult=0.01, appearance_lr_final=0.0005, appearance_lr_init=0.05, appearance_lr_max_steps=30000, checkpoint_iterations=[], compute_cov3D_python=False, convert_SHs_python=False, data_device='cuda', debug=False, debug_from=-1, densify_grad_threshold=0.0002, detect_anomaly=False, ds=1, eval=False, feat_dim=32, feature_lr=0.0075, gpu='0', images='images', ip='127.0.0.1', iterations=3000, lambda_dssim=0.2, lod=55, lowpoly=False, min_opacity=0.005, mlp_color_lr_delay_mult=0.01, mlp_color_lr_final=5e-05, mlp_color_lr_init=0.008, mlp_color_lr_max_steps=30000, mlp_cov_lr_delay_mult=0.01, mlp_cov_lr_final=0.004, mlp_cov_lr_init=0.004, mlp_cov_lr_max_steps=30000, mlp_featurebank_lr_delay_mult=0.01, mlp_featurebank_lr_final=1e-05, mlp_featurebank_lr_init=0.01, mlp_featurebank_lr_max_steps=30000, mlp_opacity_lr_delay_mult=0.01, mlp_opacity_lr_final=2e-05, mlp_opacity_lr_init=0.002, mlp_opacity_lr_max_steps=30000, model_path='C://Users//lenovo//Desktop//XAG//daima//youhua//Scaffold-GS//output', n_offsets=10, offset_lr_delay_mult=0.01, offset_lr_final=0.0001, offset_lr_init=0.01, offset_lr_max_steps=30000, opacity_lr=0.02, percent_dense=0.01, port=6009, position_lr_delay_mult=0.01, position_lr_final=0.0, position_lr_init=0.0, position_lr_max_steps=30000, quiet=False, ratio=1, resolution=-1, rotation_lr=0.002, save_iterations=[30000, 3000], scaling_lr=0.007, sh_degree=3, source_path='C://Users//lenovo//Desktop//XAG//daima//youhua//Scaffold-GS//data//data_train//scene2', start_checkpoint=None, start_stat=500, success_threshold=0.8, test_iterations=[30000], undistorted=False, update_depth=3, update_from=1500, update_hierachy_factor=4, update_init_factor=16, update_interval=100, update_until=15000, use_feat_bank=False, use_wandb=False, voxel_size=0.001, warmup=False, white_background=False)
2024-05-28 20:25:56,930 - INFO: using GPU 0
2024-05-28 20:25:56,932 - INFO: save code failed~
2024-05-28 20:25:56,932 - INFO: Optimizing C://Users//lenovo//Desktop//XAG//daima//youhua//Scaffold-GS//output
Tensorboard not available: not logging progress [28/05 20:25:56]
Reading camera 251/251 [28/05 20:25:57]
start fetching data from ply file [28/05 20:25:57]
Loading Training Cameras [28/05 20:25:57]
Loading Test Cameras [28/05 20:26:01]
Initial voxel_size: 0.001 [28/05 20:26:01]
Number of points at initialisation :  131848 [28/05 20:26:01]
Training progress: 100%|██████████| 3000/3000 [02:58<00:00, 16.83it/s, Loss=0.0721909]
2024-05-28 20:28:59,648 - INFO: 
[ITER 3000] Saving Gaussians
2024-05-28 20:29:01,758 - INFO: 
Training complete.
2024-05-28 20:29:01,758 - INFO: 
Starting Rendering~
Loading trained model at iteration 3000 [28/05 20:29:01]
Reading camera 251/251 [28/05 20:29:02]
start fetching data from ply file [28/05 20:29:02]
Loading Training Cameras [28/05 20:29:02]
Loading Test Cameras [28/05 20:29:05]
Rendering progress: 0it [00:00, ?it/s]
2024-05-28 20:29:05,409 - INFO: Test FPS: nan
2024-05-28 20:29:05,411 - INFO: 
Rendering complete.
2024-05-28 20:29:05,411 - INFO: 
 Starting evaluation...
Metric evaluation progress: 0it [00:00, ?it/s]
2024-05-28 20:29:05,411 - INFO: model_paths: C://Users//lenovo//Desktop//XAG//daima//youhua//Scaffold-GS//output
2024-05-28 20:29:05,412 - INFO:   SSIM :          nan
2024-05-28 20:29:05,412 - INFO:   PSNR :          nan
2024-05-28 20:29:05,412 - INFO:   LPIPS:          nan
2024-05-28 20:29:05,413 - INFO: 
Evaluating complete.
 [28/05 20:29:05]
 [28/05 20:29:05]

**Comment by inspirelt** [comment]
Could you provide your data file structure and your training script?

**Comment by gyxw521** [comment]
The data file structure:
C:/Users/lenovo/Desktop/XAG/daima/youhua/Scaffold-GS/data/
├── dataset_chaye
│   ├── scene1/
│   │   ├── images
│   │   │   ├── IMG_0.jpg
│   │   │   ├── IMG_1.jpg
│   │   │   ├── ...
│   │   ├── sparse/
│   │       └──0/

The training script is use you provide train.py

**Comment by tongji-rkr** [comment]
Have you tried using viewer? Can you show me the result in viewer

**Comment by gyxw521** [comment]
viewer is useful,only can't test,the test result is Nan

**Comment by MulinYu** [comment]
Adding '--eval' in the command line.

**Comment by ihuchao** [comment]
"Adding '--eval' in the command line." is tested effectively!
